File: app/src/main/java/com/igordanilchik/daggertest/RepositoryModule.java
package com.igordanilchik.daggertest;

import com.igordanilchik.daggertest.data.Repository;

import java.io.File;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;

@Module
class RepositoryModule {

    @Provides
    Repository provideRepository(File cacheDir, OkHttpClient httpClient) {
        return new Repository(cacheDir, httpClient);
    }
}

File: app/src/main/java/com/igordanilchik/daggertest/common/di/RepositoryModule.kt
package com.igordanilchik.daggertest.common.di

import com.igordanilchik.daggertest.data.source.IRepository
import com.igordanilchik.daggertest.data.source.Repository
import com.igordanilchik.daggertest.data.source.local.ILocalDataSource
import com.igordanilchik.daggertest.data.source.remote.IRemoteDataSource
import com.igordanilchik.daggertest.dto.ICatalogueMapper
import dagger.Module
import dagger.Provides

@Module
class RepositoryModule {

    @Provides
    internal fun repository(
            localDataSource: ILocalDataSource,
            remoteDataSource: IRemoteDataSource,
            mapper: ICatalogueMapper
    ): IRepository =
            Repository(
                    localDataSource,
                    remoteDataSource,
                    mapper
            )
}
